qa/rpc-tests/wallet_sapling.py

```diff
@@ -19,15 +19,38 @@ class WalletSaplingTest(BitcoinTestFramework):
     def setup_nodes(self):
         return start_nodes(4, self.options.tmpdir, [[
             '-nuparams=5ba81b19:201', # Overwinter
-            '-nuparams=76b809bb:201', # Sapling
+            '-nuparams=76b809bb:202', # Sapling
         ]] * 4)
 
     def run_test(self):
         # Sanity-check the test harness
         assert_equal(self.nodes[0].getblockcount(), 200)
 
+        # Verify RPCs disallow Sapling value transfer if Sapling is not active
+        tmp_taddr = self.nodes[3].getnewaddress()
+        tmp_zaddr = self.nodes[3].z_getnewaddress('sapling')
+        try:
+            recipients = []
+            recipients.append({"address": tmp_zaddr, "amount": Decimal('20')})
+            self.nodes[3].z_sendmany(tmp_taddr, recipients, 1, 0)
+            raise AssertionError("Should have thrown an exception")
+        except JSONRPCException as e:
+            assert_equal("Invalid parameter, Sapling has not activated", e.error['message'])
+        try:
+            recipients = []
+            recipients.append({"address": tmp_taddr, "amount": Decimal('20')})
+            self.nodes[3].z_sendmany(tmp_zaddr, recipients, 1, 0)
+            raise AssertionError("Should have thrown an exception")
+        except JSONRPCException as e:
+            assert_equal("Invalid parameter, Sapling has not activated", e.error['message'])
+        try:
+            self.nodes[3].z_shieldcoinbase(tmp_taddr, tmp_zaddr)
+            raise AssertionError("Should have thrown an exception")
+        except JSONRPCException as e:
+            assert_equal("Invalid parameter, Sapling has not activated", e.error['message'])
+
         # Activate Sapling
-        self.nodes[2].generate(1)
+        self.nodes[2].generate(2)
```

Comment:
This tests that Sapling addresses are disallowed before Overwinter, but not that they are disallowed after Overwinter and before Sapling.

Reply:
I pushed an update which generates a block so the test occurs when Overwinter is active, to mimic mainnet.